Configuring with a mix of GNU and non-GNU tools seems to work just fine.

See: https://github.com/MoarVM/MoarVM/issues/1149

--- Configure.pl.orig
+++ Configure.pl
@@ -192,14 +192,14 @@
     if ($gnu_toolchain && $gnu_count != scalar @check_tools) {
         print "\nNot all tools in the toolchain are GNU. Please correct this and retry.\n"
             . "See README.markdown for more details.\n\n";
-        exit -1;
+#       exit -1;
     }
 
     ## Otherwise, make sure that none of them are GNU
     elsif (!$gnu_toolchain && $gnu_count != 0) {
         print "\nGNU tools detected, despite this not being a GNU-oriented build.\n"
             ." Please correct this and retry. See README.markdown for more details.\n\n";
-        exit -1;
+#       exit -1;
     }
 }
 
